Warning while building Node.js

We should probably add -Wunused-function flag to our build

```
[1450/4824] CXX obj/deps/v8/third_party/simdutf/simdutf.simdutf.o
../../deps/v8/third_party/simdutf/simdutf.cpp:13596:28: warning: unused function 'low_surrogate' [-Wunused-function]
 13596 | simdutf_really_inline bool low_surrogate(char16_t c) {
       |                            ^~~~~~~~~~~~~
../../deps/v8/third_party/simdutf/simdutf.cpp:14446:15: warning: unused function 'tail_encode_base64' [-Wunused-function]
 14446 | inline size_t tail_encode_base64(char *dst, const char *src, size_t srclen,
       |               ^~~~~~~~~~~~~~~~~~
../../deps/v8/third_party/simdutf/simdutf.cpp:25086:30: warning: unused function 'count_code_points_bytemask' [-Wunused-function]
 25086 | simdutf_really_inline size_t count_code_points_bytemask(const char *in,
       |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
```

--- a/include/simdutf/common_defs.h
+++ b/include/simdutf/common_defs.h
@@ -24,6 +24,20 @@
   #define simdutf_log_assert(cond, msg)
 #endif
 
+#if SIMDUTF_CPLUSPLUS17
+#define simdutf_unused [[maybe_unused]]
+#endif // SIMDUTF_CPLUSPLUS17
+
+#if SIMDUTF_CPLUSPLUS20
+#if defined(__has_cpp_attribute)
+  #if __has_cpp_attribute(likely) >= 201803L && __has_cpp_attribute(unlikely) >= 201803L
+    #define simdutf_likely [[likely]]
+    #define simdutf_unlikely [[unlikely]]
+  #endif // __has_cpp_attribute(likely)
+#endif // __has_cpp_attribute
+#endif // SIMDUTF_CPLUSPLUS20
+
+
 #if defined(SIMDUTF_REGULAR_VISUAL_STUDIO)
   #define SIMDUTF_DEPRECATED __declspec(deprecated)
 
@@ -31,7 +45,9 @@
   #define simdutf_always_inline __forceinline // always inline, no matter what
   #define simdutf_never_inline __declspec(noinline)
 
+  #ifndef simdutf_unused
   #define simdutf_unused
+  #endif // simdutf_unused
   #define simdutf_warn_unused
 
   #ifndef simdutf_likely
@@ -74,8 +90,9 @@
     inline __attribute__((always_inline)) // always inline, no matter what
   #define SIMDUTF_DEPRECATED __attribute__((deprecated))
   #define simdutf_never_inline inline __attribute__((noinline))
-
+  #ifndef simdutf_unused
   #define simdutf_unused __attribute__((unused))
+  #endif // simdutf_unused
   #define simdutf_warn_unused __attribute__((warn_unused_result))
 
   #ifndef simdutf_likely

--- a/include/simdutf/common_defs.h
+++ b/include/simdutf/common_defs.h
@@ -25,18 +25,8 @@
 #endif
 
 #if SIMDUTF_CPLUSPLUS17
-#define simdutf_unused [[maybe_unused]]
+  #define simdutf_unused [[maybe_unused]]
 #endif // SIMDUTF_CPLUSPLUS17
-
-#if SIMDUTF_CPLUSPLUS20
-#if defined(__has_cpp_attribute)
-  #if __has_cpp_attribute(likely) >= 201803L && __has_cpp_attribute(unlikely) >= 201803L
-    #define simdutf_likely [[likely]]
-    #define simdutf_unlikely [[unlikely]]
-  #endif // __has_cpp_attribute(likely)
-#endif // __has_cpp_attribute
-#endif // SIMDUTF_CPLUSPLUS20
-
 
 #if defined(SIMDUTF_REGULAR_VISUAL_STUDIO)
   #define SIMDUTF_DEPRECATED __declspec(deprecated)
@@ -46,7 +36,7 @@
   #define simdutf_never_inline __declspec(noinline)
 
   #ifndef simdutf_unused
-  #define simdutf_unused
+    #define simdutf_unused
   #endif // simdutf_unused
   #define simdutf_warn_unused
 
@@ -91,7 +81,7 @@
   #define SIMDUTF_DEPRECATED __attribute__((deprecated))
   #define simdutf_never_inline inline __attribute__((noinline))
   #ifndef simdutf_unused
-  #define simdutf_unused __attribute__((unused))
+    #define simdutf_unused __attribute__((unused))
   #endif // simdutf_unused
   #define simdutf_warn_unused __attribute__((warn_unused_result))
 

--- a/include/simdutf/scalar/base64.h
+++ b/include/simdutf/scalar/base64.h
@@ -644,9 +644,8 @@ simdutf_constexpr23 size_t tail_encode_base64_impl(
 
 // Returns the number of bytes written. The destination buffer must be large
 // enough. It will add padding (=) if needed.
-simdutf_unused inline simdutf_constexpr23 size_t
-tail_encode_base64(char *dst, const char *src, size_t srclen,
-                   base64_options options) {
+simdutf_unused inline simdutf_constexpr23 size_t tail_encode_base64(
+    char *dst, const char *src, size_t srclen, base64_options options) {
   return tail_encode_base64_impl(dst, src, srclen, options);
 }
 

--- a/include/simdutf/scalar/base64.h
+++ b/include/simdutf/scalar/base64.h
@@ -644,9 +644,9 @@ simdutf_constexpr23 size_t tail_encode_base64_impl(
 
 // Returns the number of bytes written. The destination buffer must be large
 // enough. It will add padding (=) if needed.
-inline simdutf_constexpr23 size_t tail_encode_base64(char *dst, const char *src,
-                                                     size_t srclen,
-                                                     base64_options options) {
+simdutf_unused inline simdutf_constexpr23 size_t
+tail_encode_base64(char *dst, const char *src, size_t srclen,
+                   base64_options options) {
   return tail_encode_base64_impl(dst, src, srclen, options);
 }
 

--- a/include/simdutf/scalar/utf16.h
+++ b/include/simdutf/scalar/utf16.h
@@ -144,12 +144,9 @@ template <endianness big_endian> constexpr bool is_low_surrogate(char16_t c) {
   return (0xdc00 <= c && c <= 0xdfff);
 }
 
-simdutf_really_inline constexpr bool high_surrogate(char16_t c) {
+simdutf_unused simdutf_really_inline constexpr bool
+high_surrogate(char16_t c) {
   return (0xd800 <= c && c <= 0xdbff);
-}
-
-simdutf_really_inline constexpr bool low_surrogate(char16_t c) {
-  return (0xdc00 <= c && c <= 0xdfff);
 }
 
 template <endianness big_endian>

--- a/include/simdutf/scalar/utf16.h
+++ b/include/simdutf/scalar/utf16.h
@@ -144,8 +144,7 @@ template <endianness big_endian> constexpr bool is_low_surrogate(char16_t c) {
   return (0xdc00 <= c && c <= 0xdfff);
 }
 
-simdutf_unused simdutf_really_inline constexpr bool
-high_surrogate(char16_t c) {
+simdutf_unused simdutf_really_inline constexpr bool high_surrogate(char16_t c) {
   return (0xd800 <= c && c <= 0xdbff);
 }
 

--- a/src/generic/utf8.h
+++ b/src/generic/utf8.h
@@ -17,8 +17,8 @@ simdutf_really_inline size_t count_code_points(const char *in, size_t size) {
 }
 
 #ifdef SIMDUTF_SIMD_HAS_BYTEMASK
-simdutf_really_inline size_t count_code_points_bytemask(const char *in,
-                                                        size_t size) {
+simdutf_unused simdutf_really_inline size_t
+count_code_points_bytemask(const char *in, size_t size) {
   using vector_i8 = simd8<int8_t>;
   using vector_u8 = simd8<uint8_t>;
   using vector_u64 = simd64<uint64_t>;

